app.py

```diff
@@ -1,23 +1,46 @@
 import json
 from flask import Flask, request, jsonify
 from flask.views import MethodView
-from werkzeug.routing import ValidationError
+from werkzeug.exceptions import HTTPException
+from HandledError import HandledError
 
 app = Flask(__name__)
+pattern = {'id': int, 'name': str, 'address': str, "work_time": str, "phone_number": str}
 with open("restaurants.json", "r") as read_file:
     data = json.load(read_file)
 
 restaurants = data['restaurants']
 
 
+def validate_post(content):
+    if content is None:
+        raise HandledError(status_code=400, description="Required data not available", name="Bad Request")
+    if content.keys() != pattern.keys():
+        raise HandledError(status_code=400, description="Sent data not correct", name="Bad Request")
+
+    if any([type(content['id']) != pattern['id'],
```

Comment:
Nice implementation! :)

There is a way to make it shorter, you can loop over pattern keys and values instead of many lines if

Reply:
fixed